src/main/java/com/example/repository/MovieRepository.java

```diff
@@ -36,10 +37,15 @@ public Movie findById(int id) {
     }
 
     public Movie findByUuid(UUID uuid) {
-        return entityManager.createQuery("SELECT m FROM Movie m WHERE m.uuid = :uuid", Movie.class).
-            setParameter("uuid", uuid).
-            getSingleResult();
+        try {
+            return entityManager.createQuery("SELECT m FROM Movie m WHERE m.uuid = :uuid", Movie.class)
+                .setParameter("uuid", uuid)
+                .getSingleResult();
+        } catch (NoResultException e) {
+            return null;
```

Comment:
If we replace "return null" with "throw new NoResultException("No movie found with UUID " + uuid);" then the standard exception mapper would catch the exception, right??